### PR TITLE
kata-ctl: Expand unit tests for CPU check

### LINKED ISSUE #5919: kata-ctl: Complete unit tests for cpu check
Expand unit tests and use table-driven testing

### PR DESCRIPTION
**Pull request by cmaf**
Change unit tests for CPU check to table-driven tests and expand test cases including temp files for cpuinfo.

Fixes #5919

Signed-off-by: Chelsea Mafrica <chelsea.e.mafrica@intel.com>

### CODE REVIEW

File: src/tools/kata-ctl/src/check.rs [hunk 1]
```diff
@@ -209,53 +217,169 @@ mod tests {
...
+                    d.result.as_ref().unwrap(),
+                    "{}",
+                    msg
+                );
+                continue;
+            }
+
+            let expected_error = format!("{}", d.result.as_ref().unwrap_err());
+            let actual_error = format!("{}", result.unwrap_err());
+            assert!(actual_error == expected_error, "{}", msg);
```
Comment (liubin) [inline src/tools/kata-ctl/src/check.rs:281]: Nit: `assert_eq!`?

File: src/tools/kata-ctl/src/check.rs [hunk 2]
```diff
@@ -209,53 +217,169 @@ mod tests {
...
```
Comment (liubin) [inline src/tools/kata-ctl/src/check.rs:343]: Ditto

**Review by liubin — APPROVED** [review]

**Review by jodh-intel — APPROVED** [review]
Thanks @cmaf - once you've rebased...

lgtm

**Pull request merged**